### PR TITLE
[INFINITY-3509] Define requested linux capabilities to be run in a Pod

### PR DESCRIPTION
**Pull request by kvish**
Working capabilities
TODO: clean up tests/add more testing, think more on how capabilities should be inputted in the svc yml

### CODE REVIEW

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluatorLinuxCapabilitiesTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        MesosResourcePool mesosResourcePool = new MesosResourcePool(offer, Optional.of(Constants.ANY_ROLE));
+
+        LinuxCapabilitiesEvaluationStage capabilityEvaluationStage = new LinuxCapabilitiesEvaluationStage(TestConstants.TASK_NAME, Optional.empty());
+        PodSpec podSpec = PodInstanceRequirementTestUtils.getCpuRequirement(1.0).getPodInstance().getPod();
+        PodInstance podInstance = new DefaultPodInstance(podSpec, 0);
+        List<String> taskNames = TaskUtils.getTaskNames(podInstance);
+
+        podInstance.getPod().getCapabilities().clear();
+
+        //TODO: how to alter the podspec in a test?
```
Comment (benclarkwood) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluatorLinuxCapabilitiesTest.java]: Use `DefaultPodSpec.newBuilder(PodSpec)`. Then you can easily alter the PodSpec.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java [hunk 2]
```diff
@@ -240,9 +241,36 @@ private static PodSpec convertPod(
                     .collect(Collectors.toList()));
         }
 
+        Collection<Protos.CapabilityInfo.Capability> linuxCapabilities = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (kvish) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java]: @benclarkwood been having some issues getting deserialization when passing a yaml list. maybe you can show me this sometime or point me to a sample

File: frameworks/helloworld/tests/test_capabilities.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        wait_for_deployment=False)
+
+    pl = sdk_plan.get_deployment_plan(config.SERVICE_NAME)
+
+    assert pl['status'] != 'COMPLETE'
+
+    marathon_config = sdk_marathon.get_config(config.SERVICE_NAME)
+
+    #make sure multiple capabilities are parsed correctly
+    marathon_config['env']['HELLO_CAPABILITIES'] = "SYS_ADMIN";
```
Comment (hectorj2f) [inline frameworks/helloworld/tests/test_capabilities.py]: @kvish Should we have a test when passing `ALL` capabilities ?

**Comment by kvish** [comment]
@hectorj2f added a test to validate ALL capabilities as well

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluatorLinuxCapabilitiesTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+        LinuxCapabilitiesEvaluationStage capabilityEvaluationStage = new LinuxCapabilitiesEvaluationStage(TestConstants.TASK_NAME, Optional.empty());
+        PodSpec podSpec = PodInstanceRequirementTestUtils.getCpuRequirement(1.0).getPodInstance().getPod();
+        PodInstance podInstance = new DefaultPodInstance(podSpec, 0);
+        List<String> taskNames = TaskUtils.getTaskNames(podInstance);
+        PodInstanceRequirement podInstanceRequirement = PodInstanceRequirement
+                .newBuilder(podInstance, taskNames)
+                .build();
+
+        //default pod should have ALL capabilities set
```
[the remark this answers is not in the document]
Comment (kvish) [reply sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluatorLinuxCapabilitiesTest.java]: this is an incorrect comment, default pod spec should NOT have ALL capabilities as this breaks existing behavior will remove that comment

[the remark this answers is not in the document]
Comment (benclarkwood) [reply sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluatorLinuxCapabilitiesTest.java]: Right, I meant more generally, I don't see you asserting that the capabilities are set correctly?

**Comment by keithchambers** [comment]
FYI for when we take this up again ... Sukesh, Artem, Ken, Todd and I worked together to define UX functionality across Marathon, Metronome and SDK. Our goal was to have a simple and consistent interface for users across these components. We agreed the desired behavior should:
- if capabilities not defined; run with default user capabilities (noop)
- if capabilities is defined; run with only the capabilities explicitly defined (including no capabilities when defined as null)

**Pull request closed without merging**